Cython/Compiler/TypeSlots.py

```diff
@@ -1096,7 +1096,7 @@ def __init__(self, old_binops):
             EmptySlot("tp_weaklist"),
             EmptySlot("tp_del"),
             EmptySlot("tp_version_tag"),
-            SyntheticSlot("tp_finalize", ["__del__"], "0", ifdef="PY_VERSION_HEX >= 0x030400a1",
+            SyntheticSlot("tp_finalize", ["__del__"], "0", ifdef="CYTHON_USE_TP_FINALIZE",
                           used_ifdef="CYTHON_USE_TP_FINALIZE"),
```

Comment:
I don't think it makes sense to use the same guard for both conditions.

Reply:
Yes this was the wrong fix. I should be applying `used_ifdef` to the `slots[]` table.